mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -186,10 +184,37 @@ def eval_fn(eval_df: Union["pd.DataFrame", "pyspark.sql.DataFrame"]) -> MetricVa
                 raw_result = model_utils.score_model_on_payload(eval_model, payload)
                 eval_result = raw_result.candidates[0].text
                 eval_result_json = json.loads(eval_result)
-                scores.append(eval_result_json["Score"])
-                justifications.append(eval_result_json["Justification"])
+                return eval_result_json["Score"], eval_result_json["Justification"]
             except Exception as e:
                 _logger.info(f"Failed to score model on payload. Error: {e!r}")
+                return None, None
+
+        scores = []
+        justifications = []
+        max_workers = 10  # You can adjust this based on your needs
+
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
+            futures = []
+            for indx, (input, output) in enumerate(zip(inputs, outputs)):
+                futures.append(
+                    executor.submit(
+                        score_model_on_one_payload,
+                        indx,
+                        input,
+                        output,
+                        variables,
+                        eval_df,
+                        evaluation_context,
+                        eval_parameters,
+                        eval_model,
+                    )
+                )
+
+            for future in as_completed(futures):
+                score, justification = future.result()
+                if score is not None and justification is not None:
```

Comment:
If some scoring fails won't the scores be mismatched with the inputs and the two lists will not be the same length - what should happen in that case?

Reply:
Good point. I've updated removed this check. `scores` and `justfications` should have None values if the judge fails. I made sure that the aggregation functions are safe against None values.